src/typed_ast.hpp

```diff
@@ -60,7 +62,7 @@ struct Declaration : public TypedAST {
 
 // doesnt have a ast_vtype
 struct DeclarationList : public TypedAST {
-	std::vector<Own<Declaration>> m_declarations;
+	std::vector<Declaration*> m_declarations;
```

Comment:
I feel like this indirection is not necessary, but I'm not gonna bother you with that now.